test/parallel/test-vm-strict-mode.js

```diff
@@ -0,0 +1,22 @@
+// https://github.com/nodejs/node/issues/5344
+'use strict';
+require('../common');
+const assert = require('assert');
+const vm = require('vm');
+const ctx = vm.createContext();
+
+vm.runInContext('w = 1;', ctx);
+assert.equal(1, ctx.w);
+
+assert.throws(function() {vm.runInContext('"use strict"; x = 1;', ctx);});
```

Comment:
Just a suggestion, but `assert.throws()` takes an optional second argument to check that the thrown error is what it is expected to be, e.g. `assert.throws(function() { … }, /ReferenceError: x is not defined/)` should work here.